elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchHSQueryImpl.java

```diff
@@ -739,6 +739,70 @@ protected void extractFacetResults() {
 		return facets;
 	}
 
+	private JsonObject buildFullTextFilter(FullTextFilterImpl fullTextFilter) {
+
+		/*
+		 * FilterKey implementations and Filter(Factory) do not have to be threadsafe wrt their parameter injection
+		 * as FilterCachingStrategy ensure a memory barrier between concurrent thread calls
+		 */
+		FilterDef def = extendedIntegrator.getFilterDefinition( fullTextFilter.getName() );
+		//def can never be null, it's guarded by enableFullTextFilter(String)
+
+		if ( isPreQueryFilterOnly( def ) ) {
+			return null;
+		}
+
+		Object filterOrFactory = createFilterInstance( fullTextFilter, def );
+		return createFullTextFilter( def, filterOrFactory );
+	}
+
+	protected JsonObject createFullTextFilter(FilterDef def, Object filterOrFactory) {
+		JsonObject jsonFilter;
+		if ( def.getFactoryMethod() != null ) {
+			try {
+				Object candidateFilter = def.getFactoryMethod().invoke( filterOrFactory );
+				if ( candidateFilter instanceof Filter ) {
+					jsonFilter = ToElasticsearch.fromLuceneFilter( (Filter) candidateFilter );
+				}
+				else if ( candidateFilter instanceof ElasticsearchFilter ) {
+					jsonFilter = GsonHolder.PARSER.parse( ( (ElasticsearchFilter) candidateFilter ).getJsonFilter() ).getAsJsonObject();
+				}
+				else {
+					throw new SearchException(
+							"Factory method does not return a Filter class or an ElasticsearchFilter class: "
+									+ def.getImpl().getName() + "." + def.getFactoryMethod().getName() );
+				}
+			}
+			catch (IllegalAccessException | InvocationTargetException e) {
+				throw new SearchException(
+						"Unable to access @Factory method: "
+								+ def.getImpl().getName() + "." + def.getFactoryMethod().getName(),
+						e );
+			}
+		}
+		else {
+			if ( filterOrFactory instanceof Filter ) {
+				jsonFilter = ToElasticsearch.fromLuceneFilter( (Filter) filterOrFactory );
+			}
+			else if ( filterOrFactory instanceof ElasticsearchFilter ) {
+				jsonFilter = GsonHolder.PARSER.parse( ( (ElasticsearchFilter) filterOrFactory ).getJsonFilter() ).getAsJsonObject();
+			}
+			else {
+				throw new SearchException(
+						"Filter implementation does not implement the Filter interface or does not extend ElasticsearchFilter: "
+								+ def.getImpl().getName() + ". "
+								+ ( def.getFactoryMethod() != null ? def.getFactoryMethod().getName() : "" ) );
+			}
+		}
+
+		if ( jsonFilter != null &&
+				( FilterCacheModeType.NONE.equals( def.getCacheMode() ) || FilterCacheModeType.INSTANCE_ONLY.equals( def.getCacheMode() ) ) ) {
+			jsonFilter.entrySet().iterator().next().getValue().getAsJsonObject().addProperty( "_cache", false );
```

Comment:
Looks like the "_cache" capabilities are deprecated:
 - https://www.elastic.co/guide/en/elasticsearch/reference/2.3/breaking_20_query_dsl_changes.html#_filter_auto_caching